Desktop release

I see you've made this app in Dart, this should make cross-platform releases much easier. I'd love to see a desktop version, specifically as a Flatpak in the future.

For building windows and linux binaries and attaching it to your release. Also, while not related, can you confirm it builds against android (flutter build apk) on flutter 3.32, main branch? I was trying to automate building APK as well but getting this error:

```txt
/home/runner/.pub-cache/hosted/pub.dev/syncfusion_flutter_pdfviewer-26.2.7/android/src/main/java/com/syncfusion/flutter/pdfviewer/SyncfusionFlutterPdfViewerPlugin.java:76: error: cannot find symbol
  public static void registerWith(io.flutter.plugin.common.PluginRegistry.Registrar registrar) {
                                                                         ^
  symbol:   class Registrar
  location: interface PluginRegistry
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':syncfusion_flutter_pdfviewer:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org/

BUILD FAILED in 5m 25s
Running Gradle task 'assembleRelease'...
```

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -99,7 +99,7 @@ jobs:
 
       - name: Zip the binary location
         run: |
-          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz -C build/linux/x64/release/bundle/ .
+          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz build/linux/x64/release/bundle/
 
       - name: Upload Linux Binary
         uses: softprops/action-gh-release@v2

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -57,17 +57,16 @@ jobs:
       - name: Build Windows App
         run: |
           flutter build windows --release
-          ls build\windows\runner\Release\
 
       - name: Zip the binary location
-        run: Compress-Archive -Path build\windows\runner\Release\ -DestinationPath build\windows\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
+        run: Compress-Archive -Path build\windows\x64\runner\Release\ -DestinationPath build\windows\x64\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
         shell: pwsh
 
       - name: Upload Windows Zip
         uses: softprops/action-gh-release@v2
         with:
           tag_name: ${{ github.event.release.tag_name }}
-          files: build/windows/runner/Release/scholarxiv-release-${{ github.event.release.tag_name }}.zip
+          files: build/windows/x64/runner/Release/scholarxiv-release-${{ github.event.release.tag_name }}.zip
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
@@ -100,9 +99,7 @@ jobs:
 
       - name: Zip the binary location
         run: |
-          mkdir -p build/linux/x64/release/bundle
-          cp -r build/linux/x64/release/bundle/* build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}
-          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz -C build/linux/x64/release/bundle
+          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz -C build/linux/x64/release/bundle/ .
 
       - name: Upload Linux Binary
         uses: softprops/action-gh-release@v2

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -1,0 +1,102 @@
+name: Build & Upload Flutter APK, Windows, and Linux Binaries
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build-android:
+    name: Build Flutter APK
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
+
+      - name: Install dependencies
+        run: flutter pub get
+
+      - name: Build Android APK
+        run: flutter build apk --release
+
+      - name: Rename the APK
+        run: |
+          mv build/app/outputs/flutter-apk/app-release.apk build/app/outputs/flutter-apk/sholarxiv-release-${{ github.event.release.tag_name }}.apk
+          echo "APK renamed to scholarxiv-release-${{ github.event.release.tag_name }}.apk"
+
+      - name: Upload APK to GitHub Release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ github.event.release.tag_name }}
+          files: build/app/outputs/flutter-apk/scholarxiv-release.apk
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-windows:
+    name: Build Windows EXE
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
+
+      - name: Enable Windows desktop
+        run: flutter config --enable-windows-desktop
+
+      - name: Install dependencies
+        run: flutter pub get
+
+      - name: Build Windows App
+        run: flutter build windows --release
+
+      - name: Zip the binary location
+        run: |
+          Compress-Archive -Path build\windows\runner\Release -DestinationPath build\windows\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
+        shell: pwsh
+
+      - name: Upload Windows Zip
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ github.event.release.tag_name }}
+          files: build/windows/runner/Release/scholarxiv-release-${{ github.event.release.tag_name }}.zip
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-linux:
+    name: Build Linux AppImage
+    runs-on: ubuntu-latest
+    container: cirrusci/flutter:latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Enable Linux desktop
+        run: flutter config --enable-linux-desktop
+
+      - name: Install dependencies
+        run: flutter pub get
+
+      - name: Build Linux Binary
+        run: flutter build linux --release
+
+      - name: Zip the binary location
+        run: |
+          mkdir -p build/linux/x64/release/bundle
+          cp -r build/linux/x64/release/bundle/* build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}
+          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz -C build/linux/x64/release/bundle
+
+      - name: Upload Linux Binary
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ github.event.release.tag_name }}
+          files: build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -59,6 +59,7 @@ jobs:
 
       - name: Zip the binary location
         run: |
+          ls build\windows\runner\Release
           Compress-Archive -Path build\windows\runner\Release -DestinationPath build\windows\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
         shell: pwsh
 
@@ -71,12 +72,16 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   build-linux:
-    name: Build Linux AppImage
+    name: Build Linux Binary
     runs-on: ubuntu-latest
-    container: cirrusci/flutter:latest
 
     steps:
       - uses: actions/checkout@v4
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
 
       - name: Enable Linux desktop
         run: flutter config --enable-linux-desktop

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -5,36 +5,36 @@ on:
     types: [created]
 
 jobs:
-  build-android:
-    name: Build Flutter APK
-    runs-on: ubuntu-latest
+  # build-android:
+  #   name: Build Flutter APK
+  #   runs-on: ubuntu-latest
 
-    steps:
-      - uses: actions/checkout@v4
+  #   steps:
+  #     - uses: actions/checkout@v4
 
-      - name: Set up Flutter
-        uses: subosito/flutter-action@v2
-        with:
-          channel: stable
+  #     - name: Set up Flutter
+  #       uses: subosito/flutter-action@v2
+  #       with:
+  #         channel: stable
 
-      - name: Install dependencies
-        run: flutter pub get
+  #     - name: Install dependencies
+  #       run: flutter pub get
 
-      - name: Build Android APK
-        run: flutter build apk --release
+  #     - name: Build Android APK
+  #       run: flutter build apk --release
 
-      - name: Rename the APK
-        run: |
-          mv build/app/outputs/flutter-apk/app-release.apk build/app/outputs/flutter-apk/sholarxiv-release-${{ github.event.release.tag_name }}.apk
-          echo "APK renamed to scholarxiv-release-${{ github.event.release.tag_name }}.apk"
+  #     - name: Rename the APK
+  #       run: |
+  #         mv build/app/outputs/flutter-apk/app-release.apk build/app/outputs/flutter-apk/sholarxiv-release-${{ github.event.release.tag_name }}.apk
+  #         echo "APK renamed to scholarxiv-release-${{ github.event.release.tag_name }}.apk"
 
-      - name: Upload APK to GitHub Release
-        uses: softprops/action-gh-release@v2
-        with:
-          tag_name: ${{ github.event.release.tag_name }}
-          files: build/app/outputs/flutter-apk/scholarxiv-release.apk
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #     - name: Upload APK to GitHub Release
+  #       uses: softprops/action-gh-release@v2
+  #       with:
+  #         tag_name: ${{ github.event.release.tag_name }}
+  #         files: build/app/outputs/flutter-apk/scholarxiv-release.apk
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   build-windows:
     name: Build Windows EXE
@@ -55,12 +55,12 @@ jobs:
         run: flutter pub get
 
       - name: Build Windows App
-        run: flutter build windows --release
+        run: |
+          flutter build windows --release
+          ls build\windows\runner\Release\
 
       - name: Zip the binary location
-        run: |
-          ls build\windows\runner\Release
-          Compress-Archive -Path build\windows\runner\Release -DestinationPath build\windows\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
+        run: Compress-Archive -Path build\windows\runner\Release\ -DestinationPath build\windows\runner\Release\scholarxiv-release-${{ github.event.release.tag_name }}.zip
         shell: pwsh
 
       - name: Upload Windows Zip
@@ -82,6 +82,12 @@ jobs:
         uses: subosito/flutter-action@v2
         with:
           channel: stable
+      
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -y && sudo apt-get upgrade -y
+          sudo apt-get install -y curl git unzip xz-utils zip libglu1-mesa
+          sudo apt-get install -y clang cmake git ninja-build pkg-config libgtk-3-dev liblzma-dev libstdc++-12-dev
 
       - name: Enable Linux desktop
         run: flutter config --enable-linux-desktop

--- a/.github/workflows/build-release-cs.yml
+++ b/.github/workflows/build-release-cs.yml
@@ -99,12 +99,12 @@ jobs:
 
       - name: Zip the binary location
         run: |
-          tar -czf build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz build/linux/x64/release/bundle/
+          tar -czf build/linux/x64/release/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz build/linux/x64/release/bundle/
 
       - name: Upload Linux Binary
         uses: softprops/action-gh-release@v2
         with:
           tag_name: ${{ github.event.release.tag_name }}
-          files: build/linux/x64/release/bundle/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz
+          files: build/linux/x64/release/scholarxiv-release-${{ github.event.release.tag_name }}.tar.gz
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

